src/odemis/acq/stream/__init__.py

```diff
@@ -266,3 +266,18 @@ def get_streams_by_type(self, stream_types):
                 streams.append(s)
 
         return streams
+
+    def get_streams_and_proj_by_type(self, stream_types):
```

Comment:
Eventually that "StreamTree" should be renamed "ProjectionTree", as its purpose is to store how the projection are displayed together. It'd still accept Streams _iff_ they have a .image. (don't do it now, but it's highlight how streams and projections are seen here).

I'd suggest to modify the behaviour of get_streams_by_type() to do what this new method does. Then, if some of the callers need really a Stream, they'll have to check themselves. From a quick check, probably all the callers are compatible with this change (but you need to double-check).